src/components/network-map-tab.js

```diff
@@ -532,62 +525,68 @@ export const NetworkMapTab = ({
-                    !refIsMapManualRefreshEnabled.current &&
-                    !isUpdatedSubstationsApplied &&
-                    updatedSubstationsIds?.length > 0
-                        ? updatedSubstationsIds
-                        : undefined;
-
-                mapEquipments
-                    .reloadImpactedSubstationsEquipments(
-                        studyUuid,
-                        currentNode,
-                        updatedSubstationsToSend,
-                        setUpdatedLines
-                    )
-                    .finally(() => setWaitingLoadEquipments(false));
-                if (updatedSubstationsToSend) {
-                    setIsUpdatedSubstationsApplied(true);
-                }
-            }
+        new MapEquipments(
+            studyUuid,
+            currentNode?.id,
+            setErrorMessage,
+            dispatch,
+            intlRef
+        );
+        dispatch(resetMapReloaded());
+    }, [currentNode, dispatch, intlRef, setErrorMessage, studyUuid]);
+
+    const updateMapEquipments = useCallback(() => {
+        if (!isNodeBuilt(currentNode) || !studyUuid || !mapEquipments) {
+            return Promise.reject();
+        }
+        console.info('Update map equipments');
+        setWaitingLoadData(true);
+        const updatedSubstationsToSend =
+            !refIsMapManualRefreshEnabled.current &&
+            !isUpdatedSubstationsApplied &&
+            updatedSubstationsIds?.length > 0
+                ? updatedSubstationsIds
+                : undefined;
+
+        if (updatedSubstationsToSend) {
+            setIsUpdatedSubstationsApplied(true);
         }
+
         dispatch(resetMapReloaded());
+
+        return mapEquipments
+            .reloadImpactedSubstationsEquipments(
+                studyUuid,
+                currentNode,
+                updatedSubstationsToSend,
+                setUpdatedLines
+            )
+            .finally(() => {
+                setWaitingLoadData(false);
+            });
     }, [
         currentNode,
         dispatch,
-        intlRef,
-        isInitialized,
         isUpdatedSubstationsApplied,
         mapEquipments,
-        setErrorMessage,
         studyUuid,
         updatedSubstationsIds,
     ]);
 
+    const updateMapEquipmentsAndGeoData = useCallback(() => {
+        updateMapEquipments().then(loadGeoData);
+    }, [updateMapEquipments, loadGeoData]);
+
     useEffect(() => {
-        if (studyUpdatedForce.eventData.headers) {
+        if (isInitialized && studyUpdatedForce.eventData.headers) {
             if (
                 studyUpdatedForce.eventData.headers[UPDATE_TYPE_HEADER] ===
                 'loadflow'
             ) {
-                //TODO reload data more intelligently
-                loadMapEquipments();
+                updateMapEquipments();
             }
         }
-    }, [studyUpdatedForce, loadMapEquipments]);
+    }, [isInitialized, studyUpdatedForce, updateMapEquipments]);
```

Comment:
`isInitialized` and `updateMapEquipments` change will proc this effect. Do we really need it ?
It' a sepcial effect of notification consumption here. We want to avoid process multiple time an identic notification.

Reply:
I added isInitialized by security but I can remove it. But updateMapEquipments I can't remove it of the deps, no ?

Reply:
it's the wrong way to do it, but it was done like this in the previous code so I suggest to keep it for now